core/toolbox/collapsible_category.js

```diff
@@ -0,0 +1,302 @@
+   * @protected
+   */
+  this.expanded_ = false;
+
+  /**
+   * The child toolbox items for this category.
+   * @type {!Array<!Blockly.ToolboxItem>}
+   * @protected
+   */
+  this.toolboxItems_ = [];
+
+  Blockly.CollapsibleToolboxCategory.superClass_.constructor.call(
+      this, categoryDef, toolbox, opt_parent);
+};
+
+Blockly.utils.object.inherits(Blockly.CollapsibleToolboxCategory, Blockly.ToolboxCategory);
+
+/**
+ * All the css class names that are used to create a collapsible
+ * category. This is all the properties from the regular category plus contents.
+ * @typedef {{
+ *            container:?string,
+ *            row:?string,
+ *            icon:?string,
+ *            label:?string,
+ *            selected:?string,
+ *            openIcon:?string,
+ *            closedIcon:?string,
+ *            contents:?string,
+ *          }}
+ */
+Blockly.CollapsibleToolboxCategory.CssConfig;
+
+/**
+ * Name used for registering a collapsible toolbox category.
+ * @const {string}
+ */
+Blockly.CollapsibleToolboxCategory.registrationName = 'collapsibleCategory';
+
+/**
+ * @override
+ */
+Blockly.CollapsibleToolboxCategory.prototype.makeDefaultCssConfig_ = function() {
+  var cssConfig = Blockly.CollapsibleToolboxCategory.superClass_.makeDefaultCssConfig_.call(this);
+  cssConfig['contents'] = 'blocklyToolboxContents';
+  return cssConfig;
+};
+
+/**
+ * Parses the contents array depending on if the category has subcategories, is a
+ * dynamic category, or if its contents are meant to be shown in the flyout.
+ * @param {!Blockly.utils.toolbox.CategoryInfo} categoryDef The information needed
+ *     to create a category.
+ * @protected
+ * TODO: Check on this.
+ * @override
+ */
+Blockly.CollapsibleToolboxCategory.prototype.parseContents_ = function(categoryDef) {
+  var contents = categoryDef['contents'];
+  var prevIsFlyoutItem = true;
+
+  if (categoryDef['custom']) {
+    this.flyoutItems_ = categoryDef['custom'];
+  } else if (contents) {
+    for (var i = 0, itemDef; (itemDef = contents[i]); i++) {
+      // Separators can exist as either a flyout item or a toolbox item so
+      // decide where it goes based on the type of the previous item.
+      if (!Blockly.registry.hasItem(Blockly.registry.Type.TOOLBOX_ITEM, itemDef['kind']) ||
+          (itemDef['kind'].toLowerCase() == Blockly.ToolboxSeparator.registrationName &&
+          prevIsFlyoutItem)) {
+        var flyoutItem = /** @type {Blockly.utils.toolbox.FlyoutItemInfo} */ (itemDef);
+        this.flyoutItems_.push(flyoutItem);
+        prevIsFlyoutItem = true;
+      } else {
+        this.createToolboxItem_(itemDef);
+        prevIsFlyoutItem = false;
+      }
+    }
+  }
+};
+
+/**
+ * Creates a toolbox item and adds it to the list of toolbox items.
+ * @param {!Blockly.utils.toolbox.ToolboxItemInfo} itemDef The information needed
+ *     to create a toolbox item.
+ * @private
+ */
+Blockly.CollapsibleToolboxCategory.prototype.createToolboxItem_ = function(itemDef) {
```

Comment:
@override ?

Reply:
I don't think this one needs an override, but I did miss it on a few others. So updated.